### PR TITLE
Enable S/MIME signed e-mail notifications

### PR DESCRIPTION
**Pull request by dregad**
This PR replaces #1512, which had merge conflicts.

The branch has been rebased on top of latest master, conflicts resolved and original commit messages improved and added a few whitespace fixes. I did not actually test this code. 

@TomekAP before this can be merged, documentation (Admin Guide docbook) should be added (in docbook/Admin_Guide/en-US/config/email.xml) and the PHPDoc comments for the new configs should be a bit more detailed (see .

I'm also wondering, did you test is the system's behavior when the config is not (or improperly) set. ?

### CODE REVIEW

**Comment by dregad** [comment]
I have tested this, and made the following changes and additions

- renamed configs from `$g_email_signing_xxx` to `$g_email_smime_xxx`, to avoid any confusion with DKIM which is also about signing
- updated PHPDoc and added new section in Admin Guide
- fixed signing not working if not using SMTP
- added new "enable" config, to provide easy on/off for the feature, and avoid calling PHPMailer::sign() if disabled.

PR also includes 2 unrelated, opportunistic changes
- move DKIM settings to their own section in Admin Guide
- $g_global_settings array now has 1 config per line, sorted alphabetically 

This is ready for merging, comments welcome.

**Comment by atrol** [comment]
> $g_global_settings array now has 1 config per line, sorted alphabetically

Can you read my mind?
I prepared the same today morning to let it become part of some cleanup PR, but wanted to wait until https://github.com/mantisbt/mantisbt/pull/1727 is merged to avoid merge conflicts

I used this snippet for it
```
sort ( $g_global_settings );
foreach ( $g_global_settings as $t_setting ) {
	echo "\t'" . $t_setting . "',\n";
}
```

I compared the output with your proposed changes and found some small differences.
Will provide line comments for that.

File: config_defaults_inc.php [hunk 1]
```diff
@@ -4301,32 +4349,100 @@
...
+	'absolute_path_default_upload_folder',
+	'admin_checks',
+	'allow_anonymous_login',
+	'allow_permanent_cookie',
+	'allow_signup',
+	'anonymous_account',
+	'bottom_include_page',
+	'bug_list_cookie',
+	'cdn_enabled',
+	'class_path','library_path',
```
Comment (atrol) [inline config_defaults_inc.php:4369]: Wrong place for `library_path`

File: config_defaults_inc.php [hunk 2]
```diff
@@ -4301,32 +4349,100 @@
...
+	'cookie_prefix',
+	'cookie_time_length',
+	'core_path',
+	'crypto_master_salt',
+	'css_include_file',
+	'css_rtl_include_file',
+	'custom_headers',
+	'database_name',
+	'db_password',
+	'db_table_prefix','db_table_suffix',
```
Comment (atrol) [inline config_defaults_inc.php:4382]: Missing newline

File: config_defaults_inc.php [hunk 3]
```diff
@@ -4301,32 +4349,100 @@
...
+	'email_smime_enable',
+	'email_smime_extracerts_file',
+	'email_smime_key_file',
+	'email_smime_key_password',
+	'fallback_language',
+	'file_type_icons',
+	'fileinfo_magic_db_file',
+	'form_security_validation',
+	'global_settings',
+	'hostname','html_valid_tags',
```
Comment (atrol) [inline config_defaults_inc.php:4403]: Missing newline between values

Comment (dregad) [reply config_defaults_inc.php]: fixed

File: config_defaults_inc.php (hunk 2 above)
Comment (dregad) [reply config_defaults_inc.php]: fixed

File: config_defaults_inc.php (hunk 1 above)
Comment (dregad) [reply config_defaults_inc.php]: fixed

**Comment by dregad** [comment]
> Can you read my mind?

Great ones think alike :wink:

Thanks for the review. I just did some search&replace in the editor, and failed to notice that in some cases there was no space after the comma, so I missed these cases you reported.

**Review by atrol — APPROVED** [review]
Looks good, didn't test.

**Comment by dregad** [comment]
Merged in a0291ffd19099e912116544726836237206b8982

**Pull request closed without merging**